react-hash-routing/client/app.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Header from './components/header';
+import Catalog from './pages/catalog';
+import NotFound from './pages/not-found';
+import ProductDetails from './pages/product-details';
+import { parseRoute } from './lib';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+  }
+
+  componentDidMount() {
+    /**
+     * Listen for hash change events on the window object
+     * Each time the window.location.hash changes, parse
+     * it with the parseRoute() function and update state
+     */
```

Comment:
The issue with having to refresh to get the pages to load is because you are missing the event listener here for hash changes on the window:
```JAVASCRIPT
window.addEventListener('hashchange', () => {
  const route = parseRoute(window.location.hash);
  this.setState({ route });
});
```